lib/diagnostic.js

```diff
@@ -6,29 +6,49 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizeWords = function normalizeWords(text) {
+  let words = text.split(/\s+/);
+  let normalizedWords = [];
+  for (let i = 0; i < words.length; i++) {
+    let word = words[i].replace(/\W+/, '');
+    normalizedWords[i] = word.toUpperCase();
+  }
 };
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function uniqueWords(text) {
+  let uniqueWordsAsKeys = {};
+  let normalizedWords = normalizeWords(text);
+  for (let i = 0; i < normalizedWords.length; i++) {
+    let word = normalizedWords[i];
+    uniqueWordsAsKeys[word] = true;
+  };
 };
```

Comment:
You need to return the keys of ```uniqueWordsAsKeys``` as an array.  Everything else in this function is fine.